xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/GatewayBalancesResult.java

```diff
@@ -0,0 +1,135 @@
+    return GatewayBalancesAssets.builder().build();
+  }
+
+  /**
+   * Amounts issued to the hotwallet addresses from the request. In the original JSON entity, the keys are addresses
+   * and the values are arrays of currency amounts they hold. The wrapping entity packages these in a
+   * {@link GatewayBalancesHotWallets} instance instead.
+   *
+   * @return The balances of the hotwallets field wrapped in a {@link GatewayBalancesHotWallets} instance.
+   */
+  @Value.Default
+  default GatewayBalancesHotWallets balances() {
+    return GatewayBalancesHotWallets.builder().build();
+  }
+
+  /**
+   * Total amounts issued to addresses not included in hotwallets. In the original JSON entity, this is presented
+   * as a map of currencies to the total value issued. The wrapping entity packages these in a
+   * {@link GatewayBalancesObligations} instance instead.
+   *
+   * @return The balances for obligations from this issuer to accounts not included in the hotwallets request
+   *   field, wrapped in a {@link GatewayBalancesObligations} instance.
+   */
+  @Value.Default
+  default GatewayBalancesObligations obligations() {
+    return GatewayBalancesObligations.builder().build();
+  }
+
+  /**
+   * The identifying hash the ledger version that was used when retrieving this data.
+   *
+   * @return An optionally-present {@link org.xrpl.xrpl4j.model.transactions.Hash256}.
+   */
+  @JsonProperty("ledger_hash")
+  Optional<Hash256> ledgerHash();
+
+  /**
+   * Get {@link #ledgerHash()}, or throw an {@link IllegalStateException} if {@link #ledgerHash()} is empty.
+   *
+   * @return The value of {@link #ledgerHash()}.
+   * @throws IllegalStateException If {@link #ledgerHash()} is empty.
+   */
+  @JsonIgnore
+  @Value.Auxiliary
+  default Hash256 ledgerHashSafe() {
+    return ledgerHash()
+      .orElseThrow(() -> new IllegalStateException("Result did not contain a ledgerHash."));
+  }
+
+  /**
+   * The ledger index of the ledger version that was used when retrieving this data.
+   *
+   * @return An optionally-present {@link LedgerIndex} representing the ledger index of the response.
+   */
+  @JsonProperty("ledger_index")
+  Optional<LedgerIndex> ledgerIndex();
+
+  /**
+   * Get {@link #ledgerIndex()}, or throw an {@link IllegalStateException} if {@link #ledgerIndex()} is empty.
+   *
+   * @return The value of {@link #ledgerIndex()}.
+   * @throws IllegalStateException If {@link #ledgerIndex()} is empty.
+   */
+  @JsonIgnore
+  @Value.Auxiliary
+  default LedgerIndex ledgerIndexSafe() {
+    return ledgerIndex()
+      .orElseThrow(() -> new IllegalStateException("Result did not contain a ledgerIndex."));
+  }
+
+  /**
+   * The ledger index of the current open ledger, which was used when retrieving this information.
+   *
+   * @return An optionally-present {@link LedgerIndex} representing the current ledger index.
+   */
+  @JsonProperty("ledger_current_index")
+  Optional<LedgerIndex> ledgerCurrentIndex();
```

Comment:
Can we add a `ledgerCurrentIndexSafe()` method as well?

Reply:
Sure.